test/oban/testing_test.exs

```diff
@@ -43,6 +43,27 @@ defmodule Oban.TestingTest do
 
       assert_enqueued worker: Ping, scheduled_at: {seconds_from_now(69), delta: 10}
     end
+
+    test "prints a helpful error message" do
+      insert_job!(%{dest: "some_node"}, worker: Ping)
+
+      try do
+        assert_enqueued worker: Ping, args: %{dest: "other_node"}
+      rescue
+        error in [ExUnit.AssertionError] ->
+          expected = """
+          Expected a job matching:
+
+          %{args: %{dest: "other_node"}, worker: Ping}
```

Comment:
I wonder if the difference between the worker names will trip people up. The expected version will typically be a proper module name, but the value from the database will always be a string.

Reply:
Yeah it might, would probably be nice to convert the module if it exists already as an atom.